### PR TITLE
fix: patch for query params in the url we need to be able to extract the type too

### PR DESCRIPTION
**Pull request by Sanix-Darker**
## WHAT

- fix the uri inferring type when there are query params on it

### CODE REVIEW

**Review by fspot — APPROVED** [review]

File: peakina/datasource.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def __post_init_post_parse__(self) -> None:
         if self.scheme not in PD_VALID_URLS:
             raise AttributeError(f"Invalid scheme {self.scheme!r}")
 
-        self.type = self.type or detect_type(self.uri, is_regex=bool(self.match))
+        self.type = self.type or detect_type(self.uri.split("?")[0], is_regex=bool(self.match))
```
Comment (lukapeschke) [inline peakina/datasource.py]: Shouldn't the URL be properly parsed instead ?

Comment (Sanix-Darker) [reply peakina/datasource.py]: Does this sounds good enought  : 
```python
from urllib.parse import urlparse
...
.....
self.type = self.type or detect_type(urlparse(self.uri).path, is_regex=bool(self.match))
```

?

Comment (lukapeschke) [reply peakina/datasource.py]: Yup looks good :+1:

**Comment by Sanix-Darker** [comment]
Will be backport to the main as soon as it got merged !

File: tests/test_datasource.py [hunk 2]
```diff
@@ -36,6 +36,7 @@ def test_type():
         DataSource("myfile.csv$")
     assert DataSource("myfile.tsv$", match=MatchEnum.GLOB).type is TypeEnum.CSV
     assert DataSource("myfile.*", match=MatchEnum.GLOB).type is None
+    assert DataSource("http://test.s3.com/myfile.csv?this=is&a=query&string=0").type is TypeEnum.CSV
```
Comment (lukapeschke) [inline tests/test_datasource.py:39]:
```suggestion
    assert DataSource("http://test.s3.com/myfile.csv?this=is&a=query&string=0").type is TypeEnum.CSV
    assert DataSource("s3://bucket/object.xlsx").type is TypeEnum.EXCEL
    assert DataSource("s3+protocol://bucket/object.json?q=x").type is TypeEnum.JSON
```

**Review by lukapeschke — APPROVED** [review]

File: tests/test_datasource.py [hunk 3]
```diff
@@ -36,6 +36,9 @@ def test_type():
...
+    assert DataSource("s3://bucket/object.xlsx").type is TypeEnum.EXCEL
+    assert DataSource("s3+protocol://bucket/object.json?q=x").type is TypeEnum.JSON
```
Comment (lukapeschke) [inline tests/test_datasource.py]:
```suggestion
    assert DataSource("s3://bucket/object.json?q=x").type is TypeEnum.JSON
```

**Pull request merged**